doc: added changelog and improved readme (#2)
igor822/gomicro: README.md

Gomicro
=======

This project has the objective of being a skeleton for new micro services projects

# What is this?

> Gomicro has the purpose to set a basic structure of a project using the [clean architecture](https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html).  

Reviewer: ```suggestion
Gomicro sets a basic structure of a project using the [clean architecture](https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html).  
```

With this skeleton you can create new APIs, where it will have [echo](https://echo.labstack.com/) installed already.

## Basic structure

* `appcontext/context.go`: is our `container` of services and object within the project.

> * `config/`: here you can define all necessary configuration your application needs to run.

Reviewer: ```suggestion
* `config/`: defines all necessary configuration that your application needs to run.
```

> * `domain/`: `domain` folder you can define all structures/entites of the project

Reviewer: i would put as this way:
```suggestion
* `domain/`: defines all entities of the project
```

> * `gateway/`: connections with external world, e.g. MongoDB, MySQL

Reviewer: ```suggestion
* `gateway/`: defines all connections with the external world, e.g. MongoDB, MySQL
```

> * `usecase/`: for any usecases and business rules of the application we define in this folder

Reviewer: ```suggestion
* `usecase/`: defines the use cases and business rules of the application
```
